athenet/sparsifying/utils/interval.py

```diff
@@ -0,0 +1,284 @@
+        else:
+            res_lower = self.lower + other
+            res_upper = self.upper + other
+        return Interval(res_lower, res_upper)
+
+    __radd__ = __add__
+
+    def __sub__(self, other):
+        """Returns difference between two intervals.
+
+        :other: Interval to be subtracted."""
+        if isinstance(other, Interval):
+            res_lower = self.lower - other.upper
+            res_upper = self.upper - other.lower
+        else:
+            res_lower = self.lower - other
+            res_upper = self.upper - other
+        return Interval(res_lower, res_upper)
+
+    def __rsub__(self, other):
+        """Returns diffeerence between number and interval.
+
+        :other: A number that self will be subtracted from."""
+        res_lower = other - self.upper
+        res_upper = other - self.lower
+        return Interval(res_lower, res_upper)
+
+    def __mul__(self, other):
+        """Returns product of two intervals.
+
+        :other: Interval to be multiplied."""
+        if isinstance(other, Interval):
+            ll = self.lower * other.lower
+            lu = self.lower * other.upper
+            ul = self.upper * other.lower
+            uu = self.upper * other.upper
+            l = T.minimum(ll, lu)
+            l = T.minimum(l, ul)
+            l = T.minimum(l, uu)
+            u = T.maximum(ll, lu)
+            u = T.maximum(u, ul)
+            u = T.maximum(u, uu)
+            return Interval(l, u)
+        else:
+            ll = self.lower * other
+            uu = self.upper * other
+            l = T.minimum(ll, uu)
+            u = T.maximum(ll, uu)
+            return Interval(l, u)
+
+    __rmul__ = __mul__
+
+    def __div__(self, other):
+        """Returns quotient of self and other.
+
+        :other: Divisor.
+
+        Note: Divisor should not contain zero."""
+        lower = self.lower
+        upper = self.upper
+        if isinstance(other, Interval):
+            o_lower = other.lower
+            o_upper = other.upper
+            A = T.switch(T.gt(o_lower, 0.0), 1, 0)  # not(b_la), b_ua
+            B = T.switch(T.gt(lower, 0.0), 1, 0)
+            C = T.switch(T.gt(upper, 0.0), 1, 0)
+            b_lb = T.or_(T.and_(A, B),
+                         T.and_(1 - A, C))
+            b_ub = T.or_(1 - T.or_(A, B),
+                         T.and_(A, 1 - C))
+            la = T.switch(A, lower, upper)
+            ua = T.switch(A, upper, lower)
+            lb = T.switch(b_lb, o_upper, o_lower)
+            ub = T.switch(b_ub, o_upper, o_lower)
+            return Interval(la / lb, ua / ub)
+        else:
+            if other > 0:
+                return Interval(lower / other, upper / other)
+            else:
+                return Interval(upper / other, lower / other)
+
+    def __rdiv__(self, other):
+        """Returns quotient of other and self.
+
+        :other: Divident.
+
+        Note: Divisor should not contain zero."""
```

Comment:
I think that I will make better use of Sphinx in all the comments in interval.py, but it will generate lots of new lines of code and it would be better to be passed in next pull request as an update. If I did all of it in this pull request, it would exceed 1000 lines of code. Do you want me to make those updates under this pull request?